### ISSUE TITLE
Student Dashboard progress of submissions

### ISSUE THREAD

**Issue by stenington** [Iridescent-CM/technovation-app#1379]
Login as a student on a team
- Start your submission if not started already
- Verify that you have submission progress in the dashboard
- Verify that it is consistent with the progress on the left side of the main submission page

**Comment by madhavitechnovation** [comment]
The the submission progress on the left side works good.

One issue I faced was that in the app description segment, the user is expected to fill in 100 words. But the number of words I typed were more and the counter on the bottom showed a lower count. For example, I had typed 18 words and it showed 11/100.

**Comment by joemsak** [comment]
@madhavitechnovation the user is expected to write *no more than* 100 words, and it only counts words that are bigger than 2 characters

**Comment by Allicolyer** [comment]
Hey @RRios00 and @joemsak I think this is looking good!

Maybe we can sort out the categories now that it is working well. 
`

**Comment by Allicolyer** [comment]
@RRios00 and @joemsak. I see that they are aligned to the curriculum, but for submission  it might make sense to make these categories align with the judging rubric. 

Start 
<ul>
<li> Honor Code <li>
<li> Team Photo </li>
</ul>

Ideation
<ul> 
<li> App Name </li>
<li> App Summary </li>
</ul>

Technical
<ul> 
<li> Source Code </li>
<li> Development Platform </li>
<li> Code Cheklist </li>
</ul>

Pitch
<ul> 
<li>Pitch Video </li>
<li>Demo Video </li>
<li>Presentation Slides </li>
</ul>

Entrepreneurship
<ul>
<li>Business Plan </li>
</ul>

@joemsak would it be hard to add "team summary" to start?  Maybe we could make it "team photo and summary?"

**Comment by Allicolyer** [comment]
@joemsak Every time I save a submission item it will take me back to the last section I was in. For instance if I am in the "marketing" section and click on app description and update it, I would expect it to bring me to the "ideation" section, but it instead brings me to the "marketing" section.

I think this is not a big deal just something I noted.

**Comment by joemsak** [comment]
@Allicolyer Could we group it by curriculum for the students and mentors, and by rubric for the judges?

Also, I fixed the issue you mentioned in your last comment

**Comment by Allicolyer** [comment]
@joemsak 

 On second review really the only difference between the curriculum and the judging rubric is that in the curriculum we say "code" instead of "technical" and "business" instead of "entrepreneurship". So really the grouping would be the same for both.

 I think sticking to the same language for both would be less confusing. What do you think?

**Comment by Allicolyer** [comment]
@joemsak also when I start a new submission it takes me to the "code" section instead of the "ideation"

**Comment by kmorton2017** [comment]
i got an "uh oh" when i tried to finish the code checklist. i only had 2 points out of 10 so I uploaded pics of my progress. i tried to add info on booleans. when i was done, i clicked another submission category but received the error page

**Comment by joemsak** [comment]
@kmorton2017 You got the error when you checked something but did not fill in the explanation. However, the bug is legitimate, and I have fixed it. You can now test by checking an item but not filling in an explanation. You should get a "cannot be blank" error on that field, instead of an "uh oh" page

**Comment by ophelieh** [comment]
I love the way the submission page looks, especially the checklist on the side! :)

I am not sure if the below was already done/fixed but here is what I noticed when I tried different things. I ran out of time so wasn't able to finish testing it all but if this wasn't done before, I'm happy to finish tomorrow.  

Name your app - I left it blank then clicked on "save this name" and it said "your progress has been saved..." - Once I am brought back to the dashboard, it shows me that the name still needs to be filled it but shouldn't it tell me something like "cannot be blank" when I try to submit without entering anything? 

Demo video - I entered "hello" and it saved it. We should make sure that people can only add a link. 
Pitch video - same as above. If you don't save an actual link, when you watch the video, it shows you this

![uploads/cd5e1a8e-c14a-4e06-90f5-1a9c77993e13/image.png](https://s3-us-west-2.amazonaws.com/prod.huboard.com/uploads%2Fcd5e1a8e-c14a-4e06-90f5-1a9c77993e13%2Fimage.png)

Screenshots - I was able to upload a 6 page PDF. Should we give specifications on what can be uploaded?  

Code Checklist  - 
  * Technical components - I only selected 2 and wrote 1 word in each
  * Database - selected 1 and entered 1 word
  * Mobile Feature - only selected 1 and entered 1 word
  * Picture of process - I only uploaded 1 then added a second one
It gave me an error message but it took me a little bit to figure out that it was because of the pictures of the process - it didn't show me anywhere what was causing the issue. Once I uploaded the second one, I was able to save. 

![uploads/f14420ec-c3b8-449a-97f3-1c3b58cead6c/image.png](https://s3-us-west-2.amazonaws.com/prod.huboard.com/uploads%2Ff14420ec-c3b8-449a-97f3-1c3b58cead6c%2Fimage.png)

**Comment by Allicolyer** [comment]
@joemsak what are we still waiting on here?

**Comment by joemsak** [comment]
@Allicolyer we got some feedback from @ophelieh that could use some review. A couple of things we should fix, a couple other things I think we can let them be. We should review asap

**Comment by joemsak** [comment]
@Allicolyer @ophelieh 

So, I think it's okay if we let students leave fields blank or put in bad data, because they should be reviewing their work and ensuring their submission looks good. Also, they may change their minds and decide to leave something blank to come back to it later.

As for the paper prototype / flowchart bug, I have now fixed that

**Comment by ophelieh** [comment]
Hi @joemsak! That sounds good. Everything else works great! :)
